corehq/form_processor/backends/sql/dbaccessors.py

```diff
@@ -609,16 +611,22 @@ def soft_delete_forms(domain, form_ids, deletion_date=None, deletion_id=None):
 
         return affected_count
 
-    @staticmethod
     @transaction.atomic
-    def _archive_unarchive_form(form, user_id, archive):
+    def _archive_unarchive_form(self, form, user_id, archive, archive_stub):
+        form_id = form.form_id
+        with get_cursor(XFormInstanceSQL) as cursor:
+            cursor.execute('SELECT archive_unarchive_form(%s, %s, %s)', [form_id, user_id, archive])
+            archive_stub.history_updated = True
```

Comment:
is this supposed to be `stub.archive_history_updated()`?

Reply:
Oops, https://github.com/dimagi/commcare-hq/pull/22470/commits/3a78517e70a3b506f12977c2111304964431b2ab